sample-app/src/test/java/com/test/EchoIT.java

```diff
@@ -0,0 +1,149 @@
+package com.test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertThat;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Primary;
+
+import com.paypal.springboot.resteasy.test.RestEasyApplicationTest;
+
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+
+/**
+ * Integration test for the {@link Echo} endpoint. This test will start the
+ * {@link Application} in an embedded container and executes real http requests
+ * against it. The responses are verified in the test.<br>
+ *
+ * {@link Echo} has a dependency to the {@link WordService}. We don't want to
+ * test the {@link WordService}, its already done by the
+ * {@link WordServiceTest}. We mock the {@link WordService} and use the mock
+ * inside the running {@link Application} on the embedded container.
+ *
+ * @author Christian Fehmer
+ *
+ */
+public class EchoIT extends RestEasyApplicationTest {
+
+    /**
+     * Mock for te {@link WordService}. Has to be static because we want to
+     * share the instance between this test and the running {@link Application}
+     * in the embedded container.
+     */
+    private static WordService wordsMock = mock(WordService.class);
+
+    public EchoIT() {
+        super(Application.class, "/sample-app/echo");
```

Comment:
I believe it should be only "/sample-app" here, and then "/echo" would be added when preparing the GET request statements